### PR TITLE
update DALIClassificationLoader to not use deprecated arguments

### LINKED ISSUE #4918: fix DALI example
## 🐛 Bug

failing DALI example after upgrading to 0.28

```
self = <pl_examples.basic_examples.dali_image_classifier.DALIClassificationLoader object at 0x7f1180070b90> | 820s

| def __len__(self):
| batch_count = self._size // (self._num_gpus * self.batch_size)
| >       last_batch = 1 if self._fill_last_batch else 0
| E       AttributeError: 'DALIClassificationLoader' object has no attribute '_fill_last_batch'
| pl_examples/basic_examples/dali_image_classifier.py:104: AttributeError
```

### To Reproduce

https://github.com/PyTorchLightning/pytorch-lightning/blob/94a9d3d2837eb962cb47ad2854569039a552f729/pl_examples/basic_examples/dali_image_classifier.py#L102-L105

### Additional context

it happened after v0.28 release
https://github.com/NVIDIA/DALI/releases/tag/v0.28.0

### PR DESCRIPTION
**Pull request by gan3sh500**
## What does this PR do?

Fixes #4918 issue due to DALI API change. Now uses LastBatchPolicy's FILL and DROP instead of passing fill_last_batch. 

## Before submitting
- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together? Otherwise, we ask you to create a separate PR for every change.
- [x] Did you make sure to update the documentation with your changes?
- [x] Did you write any new necessary tests? 
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:  

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified; _Bugfixes should be including in bug-fix release milestones (m.f.X) and features should be included in (m.X.b) releases._
 

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: pl_examples/basic_examples/dali_image_classifier.py [hunk 1]
```diff
@@ -93,15 +94,16 @@ def __init__(
...
             last_batch_padded=False,
     ):
-        super().__init__(pipelines, size, reader_name, auto_reset, fill_last_batch, dynamic_shape, last_batch_padded)
+        super().__init__(pipelines, size, reader_name, auto_reset, dynamic_shape,
+                         last_batch_policy=last_batch_policy, last_batch_padded=last_batch_padded)
 
     def __len__(self):
         batch_count = self._size // (self._num_gpus * self.batch_size)
-        last_batch = 1 if self._fill_last_batch else 0
+        last_batch = 0 if self._last_batch_policy == LastBatchPolicy.DROP else 1
```
Comment (Borda) [inline pl_examples/basic_examples/dali_image_classifier.py:106]: will this work also with older DALI versions, if not can we add a simple switch to something for `dali.__version__ > 0.28` and the old code for `<=0.27`

Comment (gan3sh500) [reply pl_examples/basic_examples/dali_image_classifier.py:106]: No it will not. I will add some check based on Dali version like you said.

Comment (gan3sh500) [reply pl_examples/basic_examples/dali_image_classifier.py]: I've pushed change for this. Tests passed except one which cancelled on its own.

File: pl_examples/basic_examples/dali_image_classifier.py [hunk 2]
```diff
@@ -97,11 +100,18 @@ def __init__(
             dynamic_shape=False,
             last_batch_padded=False,
     ):
-        super().__init__(pipelines, size, reader_name, auto_reset, fill_last_batch, dynamic_shape, last_batch_padded)
+        if DALI_VERSION >= '0.28.0':
```
Comment (Borda) [inline pl_examples/basic_examples/dali_image_classifier.py:103]: lets rather use `from distutils.version import LooseVersion`

Comment (gan3sh500) [reply pl_examples/basic_examples/dali_image_classifier.py]: I've added the change.

**Comment by tchaton** [comment]
Hey @gan3sh500 @Borda , ready for review ?

**Review by Borda — APPROVED** [review]

**Comment by gan3sh500** [comment]
@tchaton It is ready. I was waiting for @Borda to confirm.

**Comment by Borda** [comment]
> @tchaton It is ready. I was waiting for @Borda to confirm.

I made just small update =)

**Comment by gan3sh500** [comment]
@Borda can I try out some other issue or is there more I should do here?

**Comment by Borda** [comment]
> @Borda can I try out some other issue or is there more I should do here?

Sure, ping me on slack, we can find good one for you...

**Review by s-rog — APPROVED** [review]

**Review by SkafteNicki — APPROVED** [review]

**Pull request merged**